src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/Registration/GssfController.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SelfServiceBundle\Controller\Registration;
+
+use Exception;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Surfnet\SamlBundle\SAML2\AuthnRequestFactory;
+use Surfnet\StepupSelfService\SamlStepupProviderBundle\Saml\AssertionAdapter;
+use Surfnet\StepupSelfService\SelfServiceBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Controls registration with Generic SAML Stepup Providers (GSSPs), yielding Generic SAML Second Factors (GSSFs).
+ */
+final class GssfController extends Controller
+{
+    /**
+     * @Template
+     * @param Request $request
+     * @param string provider
+     * @return array|Response
+     */
+    public function initiateAction(Request $request, $provider)
+    {
+        $provider = $this->getProvider($provider);
+
+        if ($request->isMethod('post')) {
```

Comment:
Question: why submit an empty form just so you can check for a 'POST' rather than just using two different URLs with two different controller actions? (They also do something different and should log that as well)